### PR TITLE
fix triton ensemble

### PR DESCRIPTION
**Pull request by junjiejiangjjj**
Signed-off-by: junjie.jiang <junjie.jiang@zilliz.com>

### CODE REVIEW

File: serve/triton/to_triton_models.py [hunk 1]
```diff
@@ -194,6 +194,36 @@ class ModelToTriton(ToTriton):
...
+        for optimize in self._model_optimize:
+            if optimize in self._obj.model.optimizes:
+                if optimize == 'onnx':
+                    succ = self._obj.save_model(optimize, self._triton_files.onnx_model_file)
+                elif optimize == 'tensorrt':
+                    succ = self._obj.save_model(optimize, self._triton_files.trt_model_file)
+                else:
+                    logger.error('Unkown optimize %s' % optimize)
+                    continue
+                self._backend = optimize
```
Comment (GuoRentong) [inline serve/triton/to_triton_models.py:212]: what if there is no any match between self._model_optimize and self._obj.model.optimizes?

Comment (GuoRentong) [reply serve/triton/to_triton_models.py:212]: and the names of 'self._model_optimize' and 'self._obj.model.optimizes' are confusing. suggest to use proper names such as self._model_format_priority, self._obj.model.supported_formats

File: serve/triton/to_triton_models.py [hunk 2]
```diff
@@ -194,6 +194,36 @@ class ModelToTriton(ToTriton):
 
     Convert model to trt, torchscript or onnx.
     '''
+    def __init__(self, op, model_root, model_name, model_optimize):
```
Comment (GuoRentong) [inline serve/triton/to_triton_models.py:197]: Please add init arg descriptions to the docstring of this class

File: serve/triton/to_triton_models.py [hunk 3]
```diff
@@ -194,6 +194,36 @@ class ModelToTriton(ToTriton):
...
+                    succ = self._obj.save_model(optimize, self._triton_files.trt_model_file)
+                else:
+                    logger.error('Unkown optimize %s' % optimize)
+                    continue
+                self._backend = optimize
+        return succ
+
+
+class EnsembleToTriton:
+    def __init__(self, dag, model_root, model_name, batch_size):
```
Comment (GuoRentong) [inline serve/triton/to_triton_models.py:217]: Please add init arg descriptions to the docstring of this class

File: serve/triton/builder.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+logger = logging.getLogger()
+
+
+class Builder:
+    '''
+    Build triton models from towhee pipeline.
+
+    In V1, we only support a chain graph for we have not traced the input and output map.
+    '''
```
Comment (GuoRentong) [inline serve/triton/builder.py:31]: Please add init arg descriptions to the docstring of this class

**Review by GuoRentong — CHANGES_REQUESTED** [review]

File: serve/triton/to_triton_models.py (hunk 1 above)
Comment (junjiejiangjjj) [reply serve/triton/to_triton_models.py:212]: > what if there is no any match between self._model_optimize and self._obj.model.optimizes?

Return errors, and stop building.

File: serve/tests/unittests/triton/test_builder.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            },
+            'end': {
+                'op_name': 'end', 'init_args': None, 'call_args': None, 'child_ids': []
+            }
+        }
+        with TemporaryDirectory(dir='./') as root:
+            builer = Builder(test_dag, root, ['tensorrt'])
+            assert builer.load() is True
+            self.assertTrue(builer.build())
+
```
Comment (GuoRentong) [inline serve/tests/unittests/triton/test_builder.py:48]: consider providing a function `build_triton_models` wrapping Builder.__init__(), load(), and build()

**Review by GuoRentong — CHANGES_REQUESTED** [review]

**Pull request merged**